app/models/magic_link.rb

```diff
@@ -0,0 +1,27 @@
+class MagicLink < ApplicationRecord
+  include RandomToken
+
+  validates :email, presence: true, format: { with: /#{EMAIL_FORMAT_REGEX}/ }
+  attribute :expires_at, default: -> { 4.hours.from_now }
+  belongs_to :token, optional: true
+
+  before_create :generate_random_token
+
+  def generate_random_token
+    self.random_token ||= random_token_for(:random_token)
+  end
+
+  def public_token_or_tokens(api: nil)
```

Comment:
Je suis peu fan de cette interface, qui va à l'encontre du duck typing.

Si on veut manipuler cette interface, on se retrouve soit à faire un check de type (anti-pattern en ruby) soit à caster en array (`Array(magic_link.public_token_or_tokens)`), donc autant renvoyer directement un array.

Je renommerai donc en `#tokens` (on s'en fout in-fine que ça soit 1 ou plusieurs tokens sur la vue)